Sphinx-build complains about issue in jinja2 dependency

I only ran into this because I'm setting up a fresh development environment. Pinning the jinja2 dependency to `Jinja2<3.1` fixes it.

```
mkdir -p build/html
sphinx-build -b html . ./build/html
Traceback (most recent call last):
  File "/opt/homebrew/bin/sphinx-build", line 5, in <module>
    from sphinx.cmd.build import main
  File "/opt/homebrew/lib/python3.10/site-packages/sphinx/cmd/build.py", line 25, in <module>
    from sphinx.application import Sphinx
  File "/opt/homebrew/lib/python3.10/site-packages/sphinx/application.py", line 43, in <module>
    from sphinx.registry import SphinxComponentRegistry
  File "/opt/homebrew/lib/python3.10/site-packages/sphinx/registry.py", line 24, in <module>
    from sphinx.builders import Builder
  File "/opt/homebrew/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 26, in <module>
    from sphinx.util import import_object, logging, progress_message, rst, status_iterator
  File "/opt/homebrew/lib/python3.10/site-packages/sphinx/util/rst.py", line 21, in <module>
    from jinja2 import Environment, environmentfilter
ImportError: cannot import name 'environmentfilter' from 'jinja2' (/opt/homebrew/lib/python3.10/site-packages/jinja2/__init__.py)
make: *** [html] Error 1
```

--- a/.github/workflows/generate-docs.yml
+++ b/.github/workflows/generate-docs.yml
@@ -6,12 +6,18 @@ on:
 jobs:
   generate:
     if: github.repository == 'zeek/zeek-docs'
-    runs-on: ubuntu-latest
-
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: ['3.7', '3.8', '3.9', '3.10']
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
       - name: Fetch Dependencies
         run: sudo pip3 install -r requirements.txt
-
       - name: Generate Docs
         run: make SPHINXOPTS="-W --keep-going"

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,5 +1,6 @@
 # Note: beware of https://github.com/readthedocs/sphinx_rtd_theme/issues/1115
 # and broken behavior with docutils 0.17
+jinja2<3.1
 pygments>=2.12.0
 docutils==0.16
 sphinx_rtd_theme==0.5.2

